kloppy/infra/serializers/tracking/pff.py

```diff
@@ -0,0 +1,491 @@
+    "DM": PositionType.DefensiveMidfield,  # Provider: DM
+    "CM": PositionType.CenterMidfield,  # Provider: CM
+    "LW": PositionType.LeftWing,  # Provider: LW
+    "RW": PositionType.RightWing,  # Provider: RW
+    "D": PositionType.CenterBack,  # Provider: D (mapped to CenterBack)
+    "CF": PositionType.Striker,  # Provider: CF
+    "M": PositionType.CenterMidfield,  # Provider: M (mapped to CenterMidfield),
+    "GK": PositionType.Goalkeeper,  # Provider: GK
+    "F": PositionType.Striker,  # Provider: CF
+}
+
+
+class GameEventType:
+    """
+    GameEventType
+
+    Type of game event - may be “FIRSTKICKOFF” (first half kick-o
+    “SECONDKICKOFF” (second half kick-off), "THIRDKICKOFF" or "FOURTHKICKOFF",
+    “END” (end of half), "G" (ball hits post, bar or corner flag and stays in play),
+    “OFF” (player off), “ON” (player on), “OTB” (on-the-ball event), “OUT” (ball
+    out-of-play), “SUB” (substitution) or “VID” (video missing).
+    """
+
+    FIRSTKICKOFF = "FIRSTKICKOFF"
+    SECONDKICKOFF = "SECONDKICKOFF"
+    THIRDKICKOFF = "THIRDKICKOFF"
+    FOURTHKICKOFF = "FOURTHKICKOFF"
+    END = "END"
+    G = "G"
+    OFF = "OFF"
+    ON = "ON"
+    OTB = "OTB"
+    OUT = "OUT"
+    SUB = "SUB"
+    VID = "VID"
+
+
+class PFF_TrackingInputs(NamedTuple):
+    meta_data: IO[bytes]
+    roster_meta_data: IO[bytes]
+    raw_data: FileLike
+
+
+class PFF_TrackingDeserializer(TrackingDataDeserializer[PFF_TrackingInputs]):
+    def __init__(
+        self,
+        limit: Optional[int] = None,
+        sample_rate: Optional[float] = None,
+        coordinate_system: Optional[Union[str, Provider]] = None,
+        only_alive: Optional[bool] = False,
+    ):
+        super().__init__(limit, sample_rate, coordinate_system)
+        self.only_alive = only_alive
+
+        self._ball_owning_team = None
+        self._ball_state = BallState.DEAD
+
+    @property
+    def provider(self) -> Provider:
+        return Provider.PFF
+
+    @classmethod
+    def _get_frame_data(
+        cls,
+        players,
+        periods,
+        ball_owning_team,
+        ball_state,
+        frame,
+    ):
+        """Gets a Frame"""
+
+        # Get Frame information
+        frame_period = frame["period"]
+        frame_id = frame["frameNum"]
+        frame_timestamp = timedelta(seconds=frame["periodElapsedTime"])
+
+        # Ball coordinates
+        ball_smoothed = frame.get("ballsSmoothed")
+        if ball_smoothed:
+            ball_x = ball_smoothed.get("x")
+            ball_y = ball_smoothed.get("y")
+            ball_z = ball_smoothed.get("z")
+
+            if ball_x is None or ball_y is None or ball_z is None:
+                ball_coordinates = None
+
+            else:
+                ball_coordinates = Point3D(
+                    x=float(ball_x) if ball_x is not None else None,
+                    y=float(ball_y) if ball_y is not None else None,
+                    z=float(ball_z) if ball_z is not None else None,
+                )
```

Comment:
If it passed the previous `if` statement, none of these coordinates can be `None`.